policy/release/tasks.rego

```diff
@@ -227,6 +227,44 @@ deny contains result if {
 	)
 }
 
+# METADATA
+# title: Task is out of date
+# description: >-
+#   The Tekton Task used is or will be out of date, upgrade to a newer version.
+#   The Task was annotated with `build.appstudio.redhat.com/expires-on` annotation
+#   marking it for deprecation and expiry by a certain date.
+# custom:
+#   short_name: deprecated
+#   failure_msg: Task %q is used by pipeline task %q will/has expired on %s.
+#   solution: >-
+#     Upgrade to a newer version of the Task.
+#   collections:
+#   - redhat
+#   depends_on:
+#   - tasks.pipeline_has_tasks
+#
+deny contains result if {
+	some att in lib.pipelinerun_attestations
+	some task in tkn.tasks(att)
+
+	annotations := tkn.task_annotations(task)
+
+	deprecated_on := annotations[_expires_on_annotation]
+	deprecated_on_rfc3339 := sprintf("%s%s", [
```

Comment:
If we care about the format of the value (which I think is a good idea), can we just use `time.parse_rfc3339_ns` instead?

The RFC3339 allows different variations (not all have 20 characters). Maybe it's a good idea to just stick with it?

(This is probably contradictory to what I said earlier that we shouldn't do any time processing in rego. I do think verifying the value is in the format we expect is something rego should do, e.g. we do it for most of the rule data values.)

Reply:
Then that's gonna fail to be parsed the by the ec-cli: https://github.com/enterprise-contract/ec-cli/blob/e86012ff8925632abe7ef41d6a11e084f5ad09ac/internal/evaluator/conftest_evaluator.go#L136

Date without a timezone is just as ambiguous as time without a timezone. I think we should just stick to well know timedate format...

Reply:
`effective_on` here is in RFC3339 format, date without time and timezone is in UTC, because that's what we defined it as

Reply:
This is all still in development. Very easy to adjust our assumption to the realities we encounter :shrug: